modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -605,6 +609,21 @@ impl LightningClientModule {
             .await
             .map_err(|e| SendPaymentError::FinalizationError(e.to_string()))?;
 
+        let mut dbtx = self.client_ctx.module_db().begin_transaction().await;
```

Comment:
What if the client crashes right before? Event emission needs to be atomic with transaction submission. I already added a version of finalize and submit tx that accepts a dbtx in a few places, maybe you already have it available here or need to pipe it through yourself.